example/actions_guards.cpp

```diff
@@ -32,11 +35,7 @@ struct actions_guards {
       return true;
     };
 
-    auto guard2 = [](int i) {
-      assert(42 == i);
-      std::cout << "guard2" << std::endl;
-      return false;
-    };
+    auto guard2 = aux::zero_wrapper(&guard2_impl);
```

Comment:
Firstly, sorry for the late feedback, got totally occupated by CppCon.

Thanks for the example, I totally see the use case and defo we wanna support it, but I don't think we want to expose anything from aux namespace, and especially such an implementation detail as zero_wrapper to the users

Instead, I think we should provide something like `sml::call` or `sml::call_wrapper` or similar

Any thoughts?

Reply:
so something like this
```cpp
template<typename... Ts>
auto call(Ts... params){
  return aux::zero_wrapper{std::forward(params)...};
}
```
should be sufficient. Further support for `noexcept(expression)` if the code is compiled in C++20 is also nice.

Reply:
I think `wrap` describes the intent here better without going into the implementation.

Reply:
I'm okay with `wrap`, just please notice that std::forward won't be able to be used to keep sml guarantee of no depenedencies

Reply:
Yes, I was just aware of that. But looking through the templates for `zero_wrapper`, all of them use copy constructor so I don't have to worry about perfect forwarding.